Tools/WebKitTestRunner/InjectedBundle/EventSendingController.cpp

```diff
@@ -461,41 +461,57 @@ JSValueRef EventSendingController::contextClick()
 void EventSendingController::textZoomIn()
 {
     auto& injectedBundle = InjectedBundle::singleton();
-    // Ensure page zoom is reset.
-    WKBundlePageSetPageZoomFactor(injectedBundle.page()->page(), 1);
+    double zoomFactor = WKBundlePageGetTextZoomFactor(injectedBundle.page()->page()) * ZoomMultiplierRatio;
 
-    double zoomFactor = WKBundlePageGetTextZoomFactor(injectedBundle.page()->page());
-    WKBundlePageSetTextZoomFactor(injectedBundle.page()->page(), zoomFactor * ZoomMultiplierRatio);
+    auto body = adoptWK(WKMutableDictionaryCreate());
+    setValue(body, "SubMessage", "ZoomPage");
+    setValue(body, "ZoomFactor", static_cast<double>(1));
+    postSynchronousPageMessage("EventSender", body);
+    setValue(body, "SubMessage", "ZoomText");
+    setValue(body, "ZoomFactor", zoomFactor);
+    postSynchronousPageMessage("EventSender", body);
 }
 
 void EventSendingController::textZoomOut()
 {
     auto& injectedBundle = InjectedBundle::singleton();
-    // Ensure page zoom is reset.
-    WKBundlePageSetPageZoomFactor(injectedBundle.page()->page(), 1);
+    double zoomFactor = WKBundlePageGetTextZoomFactor(injectedBundle.page()->page()) / ZoomMultiplierRatio;
 
-    double zoomFactor = WKBundlePageGetTextZoomFactor(injectedBundle.page()->page());
-    WKBundlePageSetTextZoomFactor(injectedBundle.page()->page(), zoomFactor / ZoomMultiplierRatio);
+    auto body = adoptWK(WKMutableDictionaryCreate());
+    setValue(body, "SubMessage", "ZoomPage");
+    setValue(body, "ZoomFactor", static_cast<double>(1));
+    postSynchronousPageMessage("EventSender", body);
+    setValue(body, "SubMessage", "ZoomText");
+    setValue(body, "ZoomFactor", zoomFactor);
+    postSynchronousPageMessage("EventSender", body);
```

Comment:
It seems unnecessary to send two sync messages; we could add `EventSenderProxy::textZoomIn` (etc) and let UI process figure out what to do.